src/Device/Driver/AR62xx.cpp

```diff
@@ -347,15 +322,19 @@ unsigned int AR62xxDevice::GenerateSetStationCommand(uint8_t *command, unsigned
 
   /** setting frequencies */
   command[command_length++] = 22;
-  command[command_length++] = GetNthByte(active_frequency_idx, 1);
-  command[command_length++] = GetNthByte(active_frequency_idx, 0);
-  command[command_length++] = GetNthByte(passive_frequency_idx, 1);
-  command[command_length++] = GetNthByte(passive_frequency_idx, 0);
-
-  /** Creating the binary value */
-  uint16_t command_crc = CRCBitwise(command, command_length);
-  command[command_length++] = GetNthByte(command_crc, 1);
-  command[command_length++] = GetNthByte(command_crc, 0);
+
+  uint16_t tmp[] = {0x00, 0x00};
+  WriteUnalignedLE16((uint16_t *)(void *)tmp, active_frequency_idx);
```

Comment:
Why write to a temporary buffer, and then copy this buffer to another buffer?

Reply:
writing from uint16 to uint8...
I'm open for better solutions on this :-)

Reply:
I have no idea what your reply means. It doesn't appear to contain an answer to my "why" question.
Write directly into the buffer where the data is supposed to be!

Reply:
Temporary buffers removed. please have a look at the new implementation which is completely different from this and fully documented :-)